arch/risc-v/include/spinlock.h

```diff
@@ -78,7 +78,7 @@
  * RISC-V architecture supports fence instruction to ensure memory ordering
  */
 
-#ifdef __LP64__
+#ifdef CONFIG_ARCH_RV64
 typedef uint64_t spinlock_t;
```

Comment:
can we change uint64_t to uintptr_t? and then remove __LP64__/CONFIG_ARCH_RV64 check.